src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs

```diff
@@ -10444,89 +10464,99 @@ public static int Invoke(ref T result, T current, int resultIndex, int curIndex)
         internal readonly struct IndexOfMinOperator<T> : IIndexOfOperator<T> where T : INumber<T>
         {
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            public static void Invoke(ref Vector128<T> result, Vector128<T> current, ref Vector128<T> resultIndex, Vector128<T> curIndex)
+            public static void Invoke(ref Vector128<T> result, Vector128<T> current, ref Vector128<T> resultIndex, Vector128<T> currentIndex)
             {
-                Vector128<T> lessThanMask = Vector128.LessThan(result, current);
+                Vector128<T> useResult = Vector128.LessThan(result, current);
                 Vector128<T> equalMask = Vector128.Equals(result, current);
 
                 if (equalMask != Vector128<T>.Zero)
                 {
-                    Vector128<T> lessThanIndexMask = IndexLessThan(resultIndex, curIndex);
+                    Vector128<T> lessThanIndexMask = IndexLessThan(resultIndex, currentIndex);
                     if (typeof(T) == typeof(float) || typeof(T) == typeof(double))
                     {
-                        lessThanMask |= (equalMask & (IsNegative(result) | ~IsNegative(current)) & lessThanIndexMask);
+                        // bool useResult = equal && ((IsNegative(result) == IsNegative(current)) ? (resultIndex < currentIndex) : IsNegative(result));
+                        Vector128<T> resultNegative = IsNegative(result);
+                        Vector128<T> sameSign = Vector128.Equals(resultNegative.AsInt32(), IsNegative(current).AsInt32()).As<int, T>();
+                        useResult |= equalMask & Vector128.ConditionalSelect(sameSign, lessThanIndexMask, resultNegative);
```

Comment:
nit: These can be `ElementwiseSelect` since we know `sameSign` is `AllBitsSet` or `Zero` per-element

Reply:
Changed. Separately, we can go through and replace all ConditionalSelects to be ElementWiseSelect as long as the mask is known to be AllBitsSet or Zero, in particular if it's the result of some comparison operation like Equal, LessThan, GreaterThan, etc., right?

Reply:
(Should we expose ElementWiseSelect on the VectorXX types if it's actually beneficial to use?)

Reply:
(And/or make the JIT recognize some cases where it knows the bit pattern is applicable and do the better thing when possible.)